src/lib/getPeerDependenciesFromRegistry.ts

```diff
@@ -22,14 +24,23 @@ async function getPeerDependenciesFromRegistry(packageMap: Index<VersionSpec>, o
     bar.render()
   }
 
-  return Object.entries(packageMap).reduce(async (accumPromise, [pkg, version]) => {
-    const dep = await packageManager.getPeerDependencies!(pkg, version)
-    if (bar) {
-      bar.tick()
-    }
-    const accum = await accumPromise
-    return { ...accum, [pkg]: dep }
-  }, {})
+  const peerDependencies: Index<Index<string>> = Object.entries(packageMap).reduce(
+    async (accumPromise, [pkg, version]) => {
+      const dep = await packageManager.getPeerDependencies!(pkg, nodeSemver.coerce(version)?.version ?? version)
```

Comment:
For the record, both `VersionSpec` and `Version` are just `string`. They are interchangeable. I'll update the type to be more accurate, but it won't actually protect any future mistakes.

And I realized why this wasn't an issue until now. It's because we are now calling `getPeerDependenciesFromRegistry` in `runLocal` and passing it the current versions from the package.json. Should I pre-transform the results of `getCurrentDependencies` before I pass it in or what would you like me to do?

Reply:
> For the record, both `VersionSpec` and `Version` are just `string`. They are interchangeable. I'll update the type to be more accurate, but it won't actually protect any future mistakes.

Yes, I understand. Let me suggest a different way to look at it though. `VersionSpec` and `Version` are *not* actually interchangeable; it's just that the type system doesn't know this. So when it comes to `VersionSpec` and `Version`, we are thrown back into a weakly typed world. It's possible to narrow `Version` with a [template literal type](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) or [branded primitive type](https://github.com/microsoft/TypeScript/wiki/FAQ#can-i-make-a-type-alias-nominal). In the mean time, we have to treat them as different types, even if this is not yet reflected in the type system.

> And I realized why this wasn't an issue until now. It's because we are now calling `getPeerDependenciesFromRegistry` in `runLocal` and passing it the current versions from the package.json. Should I pre-transform the results of `getCurrentDependencies` before I pass it in or what would you like me to do?

We could read the installed versions, as `getPeerDependencies` did, although I don't love the reliance on node_modules. Or we could use `semver.minVersion` to take the exact version at the bottom of the range, though this has the risk of getting the incorrect peerDependency if it has changed within the range. Maybe read from disk and fallback to `semver.minVersion` if the dependency is not installed. Open to your thoughts.

Reply:
I think since this library uses package.json as the source of truth for most things, it should probably take the versions from package.json and then do minVersion to remove any ranges and use that. That feels most appropriate.